File: server/model/group.go
package model

import "time"

type Group struct {
	ID        string    `json:"id"`
	Name      string    `json:"name"`
	ParentID  string    `json:"parent_id,omitempty"`
	Icon      string    `json:"icon"`
	SortOrder int       `json:"sort_order"`
	CreatedAt time.Time `json:"created_at"`
}

type GroupCreateRequest struct {
	Name     string `json:"name"`
	ParentID string `json:"parent_id,omitempty"`
	Icon     string `json:"icon,omitempty"`
}

type GroupUpdateRequest struct {
	Name     *string `json:"name,omitempty"`
	ParentID *string `json:"parent_id,omitempty"`
	Icon     *string `json:"icon,omitempty"`
}

File: web/src/types/group.ts
export interface Group {
  id: string
  name: string
  parent_id?: string
  icon: string
  sort_order: number
  created_at: string
}

export interface GroupCreateRequest {
  name: string
  parent_id?: string
  icon?: string
}

export interface GroupUpdateRequest {
  name?: string
  parent_id?: string
  icon?: string
}
